PWGCF/EbyEFluctuations/Tasks/Diff_pT_fluct_PID.cxx

```diff
@@ -0,0 +1,758 @@
+      Subsample[i][2] = std::get<std::shared_ptr<TProfile2D>>(histos.add(Form("Subsample_%d/fA_kaon", i), "", {HistType::kTProfile2D, {axisCent, axisPt}}));
+      Subsample[i][3] = std::get<std::shared_ptr<TProfile2D>>(histos.add(Form("Subsample_%d/fA_proton", i), "", {HistType::kTProfile2D, {axisCent, axisPt}}));
+
+      Subsample[i][4] = std::get<std::shared_ptr<TProfile2D>>(histos.add(Form("Subsample_%d/fB1_hadron", i), "", {HistType::kTProfile2D, {axisCent, axis1Bin}}));
+      Subsample[i][5] = std::get<std::shared_ptr<TProfile2D>>(histos.add(Form("Subsample_%d/fB1_pion", i), "", {HistType::kTProfile2D, {axisCent, axis1Bin}}));
+      Subsample[i][6] = std::get<std::shared_ptr<TProfile2D>>(histos.add(Form("Subsample_%d/fB1_kaon", i), "", {HistType::kTProfile2D, {axisCent, axis1Bin}}));
+      Subsample[i][7] = std::get<std::shared_ptr<TProfile2D>>(histos.add(Form("Subsample_%d/fB1_proton", i), "", {HistType::kTProfile2D, {axisCent, axis1Bin}}));
+
+      Subsample[i][8] = std::get<std::shared_ptr<TProfile2D>>(histos.add(Form("Subsample_%d/fB2_hadron", i), "", {HistType::kTProfile2D, {axisCent, axis1Bin}}));
+      Subsample[i][9] = std::get<std::shared_ptr<TProfile2D>>(histos.add(Form("Subsample_%d/fB2_pion", i), "", {HistType::kTProfile2D, {axisCent, axis1Bin}}));
+      Subsample[i][10] = std::get<std::shared_ptr<TProfile2D>>(histos.add(Form("Subsample_%d/fB2_kaon", i), "", {HistType::kTProfile2D, {axisCent, axis1Bin}}));
+      Subsample[i][11] = std::get<std::shared_ptr<TProfile2D>>(histos.add(Form("Subsample_%d/fB2_proton", i), "", {HistType::kTProfile2D, {axisCent, axis1Bin}}));
+
+      Subsample[i][12] = std::get<std::shared_ptr<TProfile2D>>(histos.add(Form("Subsample_%d/fC_hadron", i), "", {HistType::kTProfile2D, {axisCent, axisPt}}));
+      Subsample[i][13] = std::get<std::shared_ptr<TProfile2D>>(histos.add(Form("Subsample_%d/fC_pion", i), "", {HistType::kTProfile2D, {axisCent, axisPt}}));
+      Subsample[i][14] = std::get<std::shared_ptr<TProfile2D>>(histos.add(Form("Subsample_%d/fC_kaon", i), "", {HistType::kTProfile2D, {axisCent, axisPt}}));
+      Subsample[i][15] = std::get<std::shared_ptr<TProfile2D>>(histos.add(Form("Subsample_%d/fC_proton", i), "", {HistType::kTProfile2D, {axisCent, axisPt}}));
+
+      Subsample[i][16] = std::get<std::shared_ptr<TProfile2D>>(histos.add(Form("Subsample_%d/fD_hadron", i), "", {HistType::kTProfile2D, {axisCent, axis1Bin}}));
+      Subsample[i][17] = std::get<std::shared_ptr<TProfile2D>>(histos.add(Form("Subsample_%d/fD_pion", i), "", {HistType::kTProfile2D, {axisCent, axis1Bin}}));
+      Subsample[i][18] = std::get<std::shared_ptr<TProfile2D>>(histos.add(Form("Subsample_%d/fD_kaon", i), "", {HistType::kTProfile2D, {axisCent, axis1Bin}}));
+      Subsample[i][19] = std::get<std::shared_ptr<TProfile2D>>(histos.add(Form("Subsample_%d/fD_proton", i), "", {HistType::kTProfile2D, {axisCent, axis1Bin}}));
```

Comment:
You will be able to simplify this a bit if you use the templated version of `add` as for instance `histos.add<TProfile2D>(....)` will provide you directly the shared pointer to store it in your vectors